### PR TITLE
Add many smarty blocks

### PR DESCRIPTION
**Pull request by julienbourdeau**

### CODE REVIEW

File: templates/checkout/_partials/cart-detailed.tpl [hunk 1]
```diff
@@ -1,9 +1,11 @@
-<div class="cart-overview js-cart" data-refresh-url="{url entity='cart' params=['ajax' => 1, 'action' => 'refresh']}">
-  <div class="body">
-    <ul>
-      {foreach from=$cart.products item=product}
-        <li class="cart-item">{include file='checkout/_partials/cart-detailed-product-line.tpl' product=$product}</li>
-      {/foreach}
-    </ul>
+{block name='cart_detailed_product_line'}
```
Comment (maximebiloe) [inline templates/checkout/_partials/cart-detailed.tpl]: block name already define inside the tpl included line 6

**Review by maximebiloe — CHANGES_REQUESTED** [review]

File: templates/customer/_partials/login-form.tpl [hunk 2]
```diff
@@ -1,30 +1,35 @@
-{include file='_partials/form-errors.tpl' errors=$errors['']}
+{block name='login_form'}
 
-{* TODO StarterTheme: HOOKS!!! *}
+  {block name='login_form'}
```
Comment (martinfojtik) [inline templates/customer/_partials/login-form.tpl]: there is `{block name='login_form'}` inside `{block name='login_form'}`

File: templates/checkout/_partials/cart-detailed-product-line.tpl [hunk 3]
```diff
@@ -1,56 +1,63 @@
...
+   >
+    {l s='Remove' d='Shop.Theme.Actions'}
+  </a>
 
-<span class="product-price">{$product.total}</span>
+  {block name='hook_cart_extra_product_actions'}
+    {hook h='displayCartExtraProductActions' product=$product}
+  {/block}
+
+  <span class="product-price">{$product.total}</span>
```
Comment (martinfojtik) [inline templates/checkout/_partials/cart-detailed-product-line.tpl]: @julienbourdeau  can you replace `{$product.total}` with this:
```
{if isset($product.is_gift) && $product.is_gift}
<span class="gift">{l s='Gift' d='Shop.Theme.Checkout'}</span>
{else}
{$product.total}
{/if}
```

File: templates/customer/_partials/login-form.tpl (hunk 2 above)
Comment (julienbourdeau) [reply templates/customer/_partials/login-form.tpl]: Thanks! I'll fix it

File: templates/checkout/_partials/cart-detailed-product-line.tpl (hunk 3 above)
Comment (julienbourdeau) [reply templates/checkout/_partials/cart-detailed-product-line.tpl]: You're right! Thanks!

**Review by maximebiloe — APPROVED** [review]

**Comment by maximebiloe** [comment]
Thank you @julienbourdeau

**Pull request merged**